Resources/Computing/HTCondorCEComputingElement.py

```diff
@@ -516,12 +522,22 @@ def __cleanup(self):
     # FIXME: again some issue with the working directory...
     # workingDirectory = self.ceParameters.get( 'WorkingDirectory', DEFAULT_WORKINGDIRECTORY )
 
+    if not self._cleanupLock.acquire(False):
+      return
+
+    now = datetime.datetime.utcnow()
+    if (self._lastCleanupTime - now).total_seconds < 60:
+      self._cleanupLock.release()
+      return
+
+    self._lastCleanupTime = now
+
     self.log.debug("Cleaning working directory: %s" % self.workingDirectory)
 
     # remove all files older than 120 minutes starting with DIRAC_ Condor will
     # push files on submission, but it takes at least a few seconds until this
     # happens so we can't directly unlink after condor_submit
-    status, stdout = commands.getstatusoutput('find %s -mmin +120 -name "DIRAC_*" -delete ' % self.workingDirectory)
+    status, stdout = commands.getstatusoutput('find -O3 %s -maxdepth 1 -mmin +120 -name "DIRAC_*" -delete ' % self.workingDirectory)
```

Comment:
This, mostly maxdepth makes the find much faster to less than a tenth of a second in my machine

Reply:
From my experience of OpenStack volumes at CERN, calling `stat` can take thousands of times longer the first time you call it so benchmarks might be unreliable if you're not remounting the volume every time.

Reply:
This is the "disk" of the VM, not an additional volume, or you mean also that? 
If we call it often enough we should keep the caches warm, no?

Reply:
Bit limiting the call to the base folder and not going into all the subfolders were there are no "DIRAC_" files will  definitely speed things up, a lot.

Reply:
I've not really investigated it (I just used a database instead) so I might have drawn the wrong conclusions but:

> This is the "disk" of the VM, not an additional volume, or you mean also that?

The "disk" is just a CEPH volume by default AFAIK. Regardless I've definitely seen this behaviour on the VM's root partition.

> If we call it often enough we should keep the caches warm, no?

Anecdotally this hasn't been enough to reliably list trees with 100,000+ files and the cache seemed to randomly reset.

Reply:
BTW just fix the length: 
```
./Resources/Computing/HTCondorCEComputingElement.py:540:121: E501 line too long (132 > 120 characters)
```

Reply:
I fixed the long line.